libraries/extractor/src/main/java/androidx/media3/extractor/DtsUtil.java

```diff
@@ -223,8 +357,262 @@ public static int getDtsFrameSize(byte[] data) {
+          frameBits.skipBits(24); // LanguageDescriptor
+        }
+        if (frameBits.readBits(1) == 1) { // bInfoTextPresent
+          int nuInfoTextByteSize = frameBits.readBits(10) + 1;
+          frameBits.skipBits(nuInfoTextByteSize * 8); // InfoTextString
+        }
+        frameBits.skipBits(5); // nuBitResolution
+
+        sampleRate = SAMPLE_RATE_EXTSS[frameBits.readBits(4)];
+        channelCount = frameBits.readBits(8) + 1;
+      }
+    }
+
+    // Number of audio sample in a compressed DTS frame
+    int sampleCount = nExSSFrameDurationCode * (sampleRate / REF_CLOCK_TABLE[ nRefClockCode]);
+    return new DtsFormatInfo(
+        MimeTypes.AUDIO_DTS_EXPRESS,
+        channelCount,
+        sampleRate,
+        extFrameSize,
+        sampleCount,
+        0);
+  }
+
+  /**
+   * Returns the size of frame header in a DTS-HD frame by parsing a few bytes of data
+   * (minimum 8 bytes) from the input bitstream(containing only Extension Sub-stream).
+   *
+   * @param frame The initial 8 bytes(minimum) of a DTS-HD frame(having only Extension Sub-stream).
+   * @return Size of the DTS-HD frame header in bytes.
+   */
+  public static int parseDtsHdHeaderSize(byte[] frame) {
+    ParsableBitArray frameBits = getNormalizedFrameHeader(frame);
+    int nuBits4Header;
+    frameBits.skipBits( 32 + 8 + 2); // SYNC, UserDefinedBits, nExtSSIndex
+    // Unpack the num of bits to be used to read header size
+    int headerSizeType = frameBits.readBits(1);
+    if (headerSizeType == 0){
+      nuBits4Header = 8;
+    }
+    else{
+      nuBits4Header = 12;
+    }
+    // Unpack the substream header size
+    return frameBits.readBits(nuBits4Header) + 1;
+  }
+
+  /**
+   * Returns the DTS format given {@code data} containing the DTS-UHD(Profile 2) frame according to
+   * ETSI TS 103 491 V1.2.1 (2019-05), Section 6.4.3.
+   *
+   * @param frame The DTS-UHD frame to parse.
+   * @return The DTS format parsed from data in the header.
+   */
+  public static DtsFormatInfo parseDtsUhdFormat(byte[] frame) throws ParserException {
+    ParsableBitArray frameBits = getNormalizedFrameHeader(frame);
+    int sampleRate = 0;
+    int sampleCount = 0;
+    boolean bSyncFrameFlag = false;
+
+    int nSyncWord = frameBits.readBits(32);
+    if (nSyncWord == SYNC_VALUE_UHD_FTOC_SYNC_BE) {
+      bSyncFrameFlag = true;
+    }
+    int[] ucTable1 = new int[] { 5, 8, 10, 12 };
+    int nFTOCPayloadinBytes = extractVarLenBitFields(frameBits, ucTable1, true) + 1;
+
+    if (bSyncFrameFlag) {
+      // fullChannelBasedMixFlag, ETSI TS 103 491 V1.2.1, Section 6.4.6.1
+      if(frameBits.readBits(1) != 1) {
+        throw ParserException.createForMalformedContainer(
+            /* message= */ "Only supports full channel mask-based audio presentation",
+            /* cause= */ null);
+      }
+
+      int nBaseDurationIndex = frameBits.readBits(2);
+      sampleCount = BASE_DURATION[nBaseDurationIndex] * (frameBits.readBits(3) + 1);
```

Comment:
`m_unFrameDuration` is defined as "The frame duration in clock periods"

Making this equal to sampleCount implies there's exactly one sample per clock period. Is this assumption correct?

Reply:
This is done in the commit [ef41526](https://github.com/androidx/media/pull/275/commits/ef415268e215719ce385f5fb0e3149679a3bd91e)